openshift/templates/site-builder.sh

```diff
@@ -0,0 +1,57 @@
+        echo "Setting up configs"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p SITE_NAME=${SITE_NAME}  -f ${templates}/mariadb/config.yaml | oc apply --overwrite=false -f -
+        echo "Setting up volume"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p POOL_NAME=${POOL_NAME} -f ${templates}/mariadb/volume.yaml | oc apply --overwrite=false -f -
+        echo "Deploying"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p SITE_NAME=${SITE_NAME} -p POOL_NAME=${POOL_NAME} -f ${templates}/mariadb/deploy.yaml | oc apply -f -
+
+        printf >&2 "\nNGINX SETUP\n"
+        echo "Setting up config"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p SITE_NAME=${SITE_NAME} -p NGINX_CONF="$(envsubst '$APP_NAME,$SITE_NAME' < ${templates}/nginx/conf/wordpress.conf | cat -)" -f ${templates}/nginx/config.yaml | oc apply -f -
+        echo "Setting up service and routes"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p SITE_NAME=${SITE_NAME} -p APP_DOMAIN=${APP_DOMAIN} -f ${templates}/nginx/deploy-route.yaml | oc apply --overwrite=false -f -
+
+        printf >&2 "\nWORDPRESS SETUP\n"
+        echo "Setting up secrets"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p SITE_NAME=${SITE_NAME} -f ${templates}/wordpress/secrets.yaml | oc apply --overwrite=false -f -
+        echo "Setting up configs"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p SITE_NAME=${SITE_NAME} -p APP_DOMAIN=${APP_DOMAIN} -p WORDPRESS_MULTISITE=0 -f ${templates}/wordpress/config.yaml | oc apply --overwrite=false -f -
+        echo "Setting up volume"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p POOL_NAME=${POOL_NAME} -f ${templates}/wordpress/volume.yaml | oc apply -f -
+        echo "Deploying"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p SITE_NAME=${SITE_NAME} -p POOL_NAME=${POOL_NAME} -f ${templates}/wordpress/deploy-nginx-wordpress.yaml | oc apply -f -
+
+        printf >&2 "\nBACKUPS SETUP\n"
+        echo "Setting up configs"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p SITE_NAME=${SITE_NAME} -f ${templates}/backups/config.yaml | oc apply -f -
+        echo "Setting up volumes"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p POOL_NAME=${POOL_NAME} -f ${templates}/backups/volume-backup.yaml | oc apply -f -
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p POOL_NAME=${POOL_NAME} -f ${templates}/backups/volume-verification.yaml | oc apply -f -
+        echo "Setting up cron conainer"
+        oc process -p ENV_NAME=${OC_DEPLOY_PROJECT} -p SITE_NAME=${SITE_NAME} -p POOL_NAME=${POOL_NAME} -f ${templates}/backups/cron.yaml | oc apply -f -
+    else
+        printf >&2 "$(oc whoami)"
+    fi
+else
+    echo >&2 'Require variables to be set, run: export OC_DEPLOY_PROJECT="test" APP_DOMAIN="des-test.apps.silver.devops.gov.bc.ca" SITE_NAME="des" POOL_NAME="pool-name"'
```

Comment:
Will every user who uses this wordpress platform want to pass in "DES"-related environment variables? Should this be made more general?

Reply:
not sure of your question here... this is a tool for us to use to create sites, not for anyone to use.